maplibre/src/winit.rs

```diff
@@ -89,6 +89,10 @@ where
                     let dt = now - last_render_time;
                     last_render_time = now;
 
+                    // FIXME: Inputs are updated in RedrawRequested which
+                    // 1. Adds a strain on the main thread, maybe we need another thread for that.
+                    // 2. Delta time is correlated to the frame rate, and a low frame rate will cause
+                    //    a lag with inputs. e.g. a button is held for a certain amount of time.
```

Comment:
1. Using a different thread is not possible for web, because the DOM/Document can only be accessed from the main thread. I'm fairly sure that this is also the case for other platforms.
2. Can we fix that?

Reply:
You are probably right, and Winit probably already does something special to handle inputs.

Another thing, it's probably nothing but I have to check the Winit documentation. Imagine you call `window_input` twice before the `RedrawRequested` event is called. Maybe that could happen with a slow in framerate. You could miss a mouse click because the key press would have been `true` then `false` in-between two frames. The solution for this could be a buffer for inputs. But that is only useful if `Event::WindowEvent` can happen twice for the same input before a `Event::RedrawRequested`

Reply:
Sounds correct. Maybe we also should not trust winit's assumptions/documentation.
Can you add this comment to https://github.com/maplibre/maplibre-rs/issues/91 ?

Reply:
I have added that comment.